lib/fog/sakuracloud/models/volume/disk.rb

```diff
@@ -38,6 +38,11 @@ def carve_hostname_on_disk(hostname)
           true
         end
 
+        def attach(server_id)
+          service.attach_disk(id, server_id)
+          true
```

Comment:
Yeah. Thats a good point. I like to have too business logic in models so i can test it without reaching the provider. This makes it easier to maintain.
Usually the request is just a thin layer on top of the request, doing encoding and parsing also. =)